src/main/java/org/truffleruby/core/support/IONodes.java

```diff
@@ -470,16 +470,15 @@ public int write(int fd, DynamicObject string) {
 
             final Rope rope = rope(string);
 
-            RopeOperations.visitBytes(rope, (bytes, offset, length) -> {
-                getContext().getThreadManager().runUntilResult(this, () -> {
-                    try {
-                        stream.write(bytes, offset, length);
-                    } catch (IOException e) {
-                        throw new JavaException(e);
-                    }
-                    return BlockingAction.SUCCESS;
-                });
-            });
+            RopeOperations.visitBytes(rope, (bytes, offset, length) ->
+              getContext().getThreadManager().runUntilResult(this, () -> {
+                try {
+                    stream.write(bytes, offset, length);
+                } catch (IOException e) {
+                    throw new JavaException(e);
+                }
+                return BlockingAction.SUCCESS;
+              }));
```

Comment:
This one is fairly hard to read.
And the indent is 2 instead of 4 spaces for some reason?

Reply:
I bungled that. I've fixed the formatting, but if you find it more readable as a statement, I can back this change out.

Reply:
Seems fine now with the proper indentation.